tstate/tstate_view.go

```diff
@@ -230,13 +230,21 @@ func (ts *TStateView) Insert(ctx context.Context, key []byte, value []byte) erro
 		pastWrites:    chunks(ts.writes, k),
 	}
 	if exists {
+		// Updating just requires a Write permission
```

Comment:
I wonder if we need to be more careful with the use of `getValue` in this function now that we aren't guaranteed to have read it...

I need to think about this a bit more.

Reply:
We can't really remove it because we need it for the exists check.